Algorithms/Mathematical-Algorithms/Power/Power.cpp

```diff
@@ -0,0 +1,51 @@
+
+//https://github.com/sahilbansal17/Competitive_Coding/
+
+//Name: Anil Khatri (@imkaka)
+//Email-Id: anil.soccer.khatri@gmail.com
+
+//It calculate the power of base^power in O(log power) time complexity.
+//It is the function you need in your templete all the time.
+
+
+#include<iostream>
+#include<cstdio>
+
+using namespace std;
+//To handle large numbers for more info -> (https://codeaccepted.wordpress.com/2014/02/15/output-the-answer-modulo-109-7/)
+#define MOD 1000000007
+
+//Fast Power - Iterative versiom
+
+long long fastPower(long long base, long long power){
+
+  long long result = 1;
+
+  while(power > 0) {
+        // Can also use (power & 1) to make code even faster
```

Comment:
You should use this then.

Reply:
I have used that in the template the function is for making understand the concept.

Okay, I will do the matrix exponentiation, Thanks